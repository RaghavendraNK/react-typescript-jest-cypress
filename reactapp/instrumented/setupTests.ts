function cov_1nrbe6xq6y() {
    var path =
        '/home/raghavendra/Documents/Github/react-typescript-jest-cypress/src/setupTests.ts';
    var hash = '61624976bc4c0f3268a508313704a486eff5dc1d';
    var global = new Function('return this')();
    var gcv = '__coverage__';
    var coverageData = {
        path:
            '/home/raghavendra/Documents/Github/react-typescript-jest-cypress/src/setupTests.ts',
        statementMap: {},
        fnMap: {},
        branchMap: {},
        s: {},
        f: {},
        b: {},
        _coverageSchema: '1a1c01bbd47fc00a2c39e90264f33305004495a9',
        hash: '61624976bc4c0f3268a508313704a486eff5dc1d',
    };
    var coverage = global[gcv] || (global[gcv] = {});

    if (!coverage[path] || coverage[path].hash !== hash) {
        coverage[path] = coverageData;
    }

    var actualCoverage = coverage[path];
    {
        // @ts-ignore
        cov_1nrbe6xq6y = function () {
            return actualCoverage;
        };
    }
    return actualCoverage;
}

cov_1nrbe6xq6y();
// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInNldHVwVGVzdHMudHMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6Ijs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7O0FBZVk7Ozs7Ozs7OztBQWZaO0FBQ0E7QUFDQTtBQUNBO0FBQ0EsT0FBTywyQkFBUCIsInNvdXJjZXNDb250ZW50IjpbIi8vIGplc3QtZG9tIGFkZHMgY3VzdG9tIGplc3QgbWF0Y2hlcnMgZm9yIGFzc2VydGluZyBvbiBET00gbm9kZXMuXG4vLyBhbGxvd3MgeW91IHRvIGRvIHRoaW5ncyBsaWtlOlxuLy8gZXhwZWN0KGVsZW1lbnQpLnRvSGF2ZVRleHRDb250ZW50KC9yZWFjdC9pKVxuLy8gbGVhcm4gbW9yZTogaHR0cHM6Ly9naXRodWIuY29tL3Rlc3RpbmctbGlicmFyeS9qZXN0LWRvbVxuaW1wb3J0ICdAdGVzdGluZy1saWJyYXJ5L2plc3QtZG9tJztcbiJdfQ==
