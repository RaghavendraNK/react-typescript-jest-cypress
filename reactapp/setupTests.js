import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-17-updated';
import sinon from 'sinon';
import { expect } from 'chai';
export const mockHistoryPush = jest.fn();

configure({ adapter: new Adapter() });
global.expect = expect;
global.sinon = sinon;

jest.mock('react-i18next', () => ({
    useTranslation: () => ({ t: (key) => key }),
}));

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useHistory: () => ({
        push: mockHistoryPush,
    }),
}));
