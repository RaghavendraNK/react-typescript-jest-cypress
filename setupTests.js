import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import sinon from 'sinon';
import { expect } from 'chai';

configure({ adapter: new Adapter() });
global.expect = expect;
global.sinon = sinon;
