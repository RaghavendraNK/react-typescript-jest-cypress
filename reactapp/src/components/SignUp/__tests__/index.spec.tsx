import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { CheckBox, Logo } from 'slick-react-ui-components';
import { SignUp } from '../index';
import { mockHistoryPush } from '../../../../setupTests';

describe('SignUp', () => {
    let wrapper: ShallowWrapper;
    beforeAll(() => {
        wrapper = shallow(<SignUp />);
    });
    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
    describe('DOM', () => {
        it('should render Logo', () => {
            const component = wrapper.find(Logo);
            expect(component).toHaveLength(1);
        });
        it('should render signup form', () => {
            expect(wrapper.find({ id: 'emailAddressTxtField' })).toHaveLength(
                1
            );
            expect(wrapper.find({ id: 'passwordTxtField' })).toHaveLength(1);
            expect(wrapper.find(CheckBox)).toHaveLength(1);
            expect(wrapper.find({ id: 'signUpBtn' })).toHaveLength(1);
            expect(wrapper.find({ id: 'signUpLink' })).toHaveLength(1);
        });
        it('should navigate to sign in', () => {
            wrapper.find({ id: 'signUpLink' }).simulate('click');
            expect(mockHistoryPush).toHaveBeenCalledWith('/signin');
        });
    });
});
