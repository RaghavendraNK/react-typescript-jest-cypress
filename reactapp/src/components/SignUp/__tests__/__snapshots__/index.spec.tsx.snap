// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp should match snapshot 1`] = `
<div
  id="signUp"
>
  <div
    className="sign-up"
  >
    <div
      className="sign-up-form"
    >
      <div
        className="logo-wrapper form-row mar-b-16"
      >
        <Component
          height={28}
          isLogoWithTitle={true}
          logoTitle="LogoTitle.svg"
          width={112}
        />
      </div>
      <div
        className="form-row mar-b-20"
      >
        <b
          id="emailAddressTxtField"
          label="en:emailaddress"
          variant="outlined"
        />
      </div>
      <div
        className="form-row mar-b-12"
      >
        <b
          id="passwordTxtField"
          label="en:newpassword"
          type="password"
          variant="outlined"
        />
      </div>
      <div
        className="form-row link mar-b-20"
      >
        <Component
          checked={true}
          text={<TermsAndConditions />}
        />
      </div>
      <div
        className="form-row"
      >
        <f
          dense={40}
          id="signUpBtn"
          label="en:signuphere"
          variant="filled"
        />
      </div>
    </div>
    <div
      className="sign-up-here-section"
    >
      <div
        className="sign-up-helper-message"
      >
        en:haveaccount
      </div>
       
      <f
        id="signUpLink"
        label="en:signinhere"
        onClick={[Function]}
        variant="link"
      />
    </div>
  </div>
</div>
`;
